R/thematic.R

```diff
@@ -109,29 +116,82 @@ thematic_off <- function() {
   invisible()
 }
 
-#' Query the current thematic theme
+#' Tools for getting and restoring global state
 #'
-#' [thematic_get()] returns information about the entire theme,
-#' whereas [thematic_get_option()] returns information about a specific
-#' option.
+#' * [thematic_with_theme()]: similar to [thematic_on()], but for an single plot.
+#' * [thematic_set_theme()]: set a given `theme` object as the current theme.
+#' * [thematic_get_theme()]: obtain the current `theme`.
+#' * [thematic_get_option()]: obtain a particular `theme` option (and provide a `default`
+#' if if no `theme` is active).
+#' * [thematic_get_mixture()]: obtain a mixture of the current `theme`'s `bg` and `fg`.
 #'
+#' @param expr R code that produces a plot.
+#' @param ... arguments passed along to [thematic_on()].
+#' @param default a default value to return in the event no thematic theme is active.
+#' @rdname theme-management
 #' @export
 #' @examples
-#' thematic_get()
-#' thematic_on("darkblue", "skyblue")
-#' thematic_get_option("bg")
 #'
-#' if (interactive()) {
-#'   scales::show_col(thematic_get_mixture(seq(0, 1, by = 0.1)))
+#' # Use thematic_with_theme() for a one-time use of thematic
+#' thematic_with_theme(
+#'   plot(1:10, col = thematic_get_option("accent"), pch = 19),
+#'   "darkblue", "skyblue", accent = "red"
+#' )
+#'
+#' # Use thematic_set_theme() if doing something more complicated
+#' # like programming on top thematic (without causing side effects)
+#' my_plot <- function(expr, las = 3, ...) {
+#'   old_theme <- thematic_on("black", "white")
+#'   on.exit(thematic_set_theme(old_theme), add = TRUE)
+#'   opts <- par(las = las)
+#'   on.exit(par(opts), add = TRUE)
+#'   # Imagine some more customization with ...
+#'   force(expr)
 #' }
+#' my_plot(plot(1:10))
 #'
-thematic_get <- function() {
+#' thematic_off()
+#' thematic_get_option("bg", "white")
+#' thematic_on(bg = "red")
+#' thematic_get_option("bg", "white")
+#' thematic_off()
+#'
+#' thematic_with_theme(
+#'   scales::show_col(thematic_get_mixture(seq(0, 1, by = 0.1))),
+#'   "darkblue", "skyblue"
+#' )
+#'
+thematic_with_theme <- function(expr, ...) {
```

Comment:
reverse order.

turn `...` into a list argument.  Like the `withr` API

Reply:
I ultimately decided against, because `expr` first makes more sense for `thematic_with_device()` and it'd be good to have consistency with that...